### PR TITLE
CloudEvent Deserialization: Ignore Unknown Properties

### PR DESCRIPTION
**Pull request by mantono**
This pull request changes the default behavior when deserializing CloudEvents from a JSON string representation. Previously, no unknown properties has been allowed, resulting in an `UnrecognizedPropertyException` when an unknown value was encountered. This can how ever be problematic when an API changes and the class that the CloudEvent data represents can have new properties.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -12,3 +12,6 @@ trim_trailing_whitespace=true
 indent_style=space
 indent_size=2
 
+[*.json]
+indent_style=space
+indent_size=2
```
Comment (6185541513) [inline .editorconfig]:
```suggestion
indent_size=4
```

Comment (6185541513) [reply .editorconfig]: I might be wrong about this, but was under the impression indent_size for json was prefered to be 4.

File: src/test/resources/CloudEventWithUnknownProperty.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "eventID": "AAABYyvMcXysEwAHAAAAAA",
+  "eventTime": "2001-06-19T14:28:36Z",
+  "schemaURL": "https://klira.io/someURL",
+  "contentType": "text/plain",
+  "extensions": {
+    "authorized": true,
+    "authorizationLevel": 1024
+  },
+  "unknownProperty": 1,
+  "data": "I’m sorry, Dave. I’m afraid I can’t do that."
```
Comment (6185541513) [inline src/test/resources/CloudEventWithUnknownProperty.json]: >   "I’m sorry, Dave. I’m afraid I can’t do that."
:red_circle: :sweat_smile:

File: .editorconfig (hunk 1 above)
Comment (mantono) [reply .editorconfig]: Oh right, you are right! I thought it was the other way around. Will fix that ;)

**Comment by mantono** [comment]
I have fixed the indent as well now (4 spaces) for JSON files.

**Pull request merged**